runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolLogReplication.java

```diff
@@ -163,10 +163,10 @@ public static LogReplicationEntryMsg overrideMetadata(LogReplicationEntryMsg mes
     }
 
     public static ResponseMsg getLeadershipResponse(
-            HeaderMsg header, boolean isLeader, String nodeId) {
+            HeaderMsg header, boolean isLeader, String nodeId, boolean isStandby) {
         LogReplication.LogReplicationLeadershipResponseMsg request = LogReplication.LogReplicationLeadershipResponseMsg
                 .newBuilder()
-                .setIsLeader(isLeader)
+                .setIsLeader(isLeader && isStandby)
```

Comment:
If it's leader but the role is not that of a standby I think we should log something. It would be nice for debugging to quickly discard if the issue is in the leadership election/lock  or is it an issue in topology provider, not telling us we're standby.

Reply:
done